BParkingNano/plugins/MuonTriggerSelector.cc

```diff
@@ -184,44 +181,52 @@ void MuonTriggerSelector::produce(edm::Event& iEvent, const edm::EventSetup& iSe
       }
 
       //save reco muon 
-      //since we do not store full muon collection => useless to save original muon index
       if(recoMuonMatching_index != -1){
 	pat::Muon recoTriggerMuonCand (muon);
 	recoTriggerMuonCand.addUserInt("trgMuonIndex", trgMuonMatching_index);
 	trgmuons_out->emplace_back(recoTriggerMuonCand);
+
+	//keep track of original muon index for SelectedMuons collection
+	isTriggerMuon[iMuo] = 1;
       }
     }
-    
 
-    
-    // code simplified loop of trg inside
-      for(const pat::Muon &mu : *muons) {
-	//election cuts
-       if (mu.pt() < ptMin_) continue;
-       if (fabs(mu.eta()) > absEtaMax_) continue;
-       if (softMuonsOnly_ && !mu.isSoftMuon(PV)) continue;
-
-       // same PV as the tag muon
-       bool SkipMuon=true;
-       for (const pat::Muon & trgmu : *trgmuons_out) {
-         if( reco::deltaR(mu,trgmu) < drTrg_cleaning_ && drTrg_cleaning_ >0 )
-             continue;
-       	 if( fabs(mu.vz()-trgmu.vz()) > dzTrg_cleaning_ && dzTrg_cleaning_ >0 )
-             continue;
-         SkipMuon=false;
-       } 
-       // needs decission: what about events without trg muon? now we SKIP them
-       if (SkipMuon)  continue;
-       
-       // build transient track
-       const reco::TransientTrack muonTT((*(mu.bestTrack())), &(*bFieldHandle)); //sara: check, why not using inner track for muons?
-       if (!muonTT.isValid()) continue;
-
-       muons_out->emplace_back(mu);
-       trans_muons_out->emplace_back(muonTT);
-    }
 
 
+    // now produce output for analysis (code simplified loop of trg inside)
+    // trigger muon + all compatible in dz with any tag
+    int muIdx = -1;
+    for(auto mu : *muons) {
```

Comment:
use `const auto&` to avoid copy, given you need the index you can also use a index-based iteration